includes/template-library/sources/local.php

```diff
@@ -786,103 +788,37 @@ public function export_multiple_templates( array $template_ids ) {
-	 *
-	 * Example return data:
-	 *
-	 * [
-	 *  0 => '/www/wp-content/uploads/elementor/tmp/5eb3a7a411d44/templates/block-2-col-marble-title.json',
-	 *  1 => '/www/wp-content/uploads/elementor/tmp/5eb3a7a411d44/templates/block-2-col-text-and-photo.json',
-	 * ]
-	 *
-	 * @since 2.9.8
-	 * @access private
-	 *
-	 * @param string $temp_path - The temporary file path to scan for template files
-	 *
-	 * @return array An array of temporary files on the filesystem
-	 */
-	private function find_temp_files( $temp_path ) {
-
-		$file_names = [];
-
-		$possible_file_names = array_diff( scandir( $temp_path ), [ '.', '..' ] );
-
-		// Find nested files in the unzipped path. This happens for example when the user imports a Template Kit.
-		foreach ( $possible_file_names as $possible_file_name ) {
-			$full_possible_file_name = $temp_path . '/' . $possible_file_name;
-			if ( is_dir( $full_possible_file_name ) ) {
-				$file_names = $file_names + $this->find_temp_files( $full_possible_file_name );
-			} else {
-				$file_names[] = $full_possible_file_name;
-			}
-		}
-
-		return $file_names;
-	}
-
 	/**
 	 * Import local template.
 	 *
 	 * Import template from a file.
 	 *
+	 * @param array $import_result a file array that should include 'name' (original file name), 'type' (mime type),
+	 * and 'tmp_name' (the path to the uploaded file) properties.
+	 * @return \WP_Error|array An array of items on success, 'WP_Error' on failure.
 	 * @since 1.0.0
 	 * @access public
 	 *
-	 * @param string $name - The file name
-	 * @param string $path - The file path
-	 *
-	 * @return \WP_Error|array An array of items on success, 'WP_Error' on failure.
 	 */
-	public function import_template( $name, $path ) {
-		if ( empty( $path ) ) {
-			return new \WP_Error( 'file_error', 'Please upload a file to import' );
-		}
-
+	public function import_template( $import_result ) {
 		$items = [];
 
-		$file_extension = pathinfo( $name, PATHINFO_EXTENSION );
+		// If the import file is a Zip file with potentially multiple JSON files
+		if ( 'application/zip' === $import_result['type'] ) {
+			// TODO: 26/1/21 - CHANGE THIS TO HANDLE ZIP FILES THROUGH THE UPLOADS MANAGER
+			$extracted_files = Plugin::$instance->uploads_manager->extract_and_validate_zip( $import_result['tmp_name'] );
 
-		if ( 'zip' === $file_extension ) {
-			if ( ! class_exists( '\ZipArchive' ) ) {
-				return new \WP_Error( 'zip_error', 'PHP Zip extension not loaded' );
-			}
-
-			$zip = new \ZipArchive();
-
-			$wp_upload_dir = wp_upload_dir();
-
-			$temp_path = $wp_upload_dir['basedir'] . '/' . self::TEMP_FILES_DIR . '/' . uniqid();
-
-			$zip->open( $path );
-
-			$valid_entries = [];
+			if ( is_wp_error( $extracted_files ) ) {
+				// Remove the temporary zip file, since it's now not necessary.
+				unlink( $import_result['tmp_name'] );
+				// Delete the temporary extraction directory, since it's now not necessary.
+				rmdir( $extracted_files['temp_extraction_directory'] );
```

Comment:
This should be handled by the `Uploads_Manager` itself

Reply:
I thought you said the uploads manager does not remove files/directories.

Reply:
Use the `Uploads_Manager` to do the deletions with a new deletion method to be added to the manager.

Reply:
Done, should be reviewed to make sure it is what you meant.